README.md

```diff
@@ -1,181 +1,93 @@
 
-### Example: custom config
-
-To configure a custom/private server, just 
-
-1. load the lspconfig module: `local lspconfig = require('lspconfig')`,
-2. Define the config: `lspconfig.configs.foo_lsp = { … }`
-3. Call `setup()`: `lspconfig.foo_lsp.setup{}`
-
-```lua
-local lspconfig = require'lspconfig'
-local configs = require'lspconfig/configs'
--- Check if it's already defined for when reloading this file.
-if not lspconfig.foo_lsp then
-  configs.foo_lsp = {
-    default_config = {
-      cmd = {'/home/ashkan/works/3rd/lua-language-server/run.sh'};
-      filetypes = {'lua'};
-      root_dir = function(fname)
-        return lspconfig.util.find_git_ancestor(fname) or vim.loop.os_homedir()
-      end;
-      settings = {};
-    };
-  }
-end
-lspconfig.foo_lsp.setup{}
-```
+Enabling most language servers is as easy as installing the language server, ensuring it is on your PATH, and adding the following to your config:
 
-### Example: override default config for all servers
-
-If you want to change default configs for all servers, you can override default_config like this. In this example, we additionally add a check for log_level and message_level which can be passed to the server to control the verbosity of "window/logMessage".
-
-```lua
-local lspconfig = require'lspconfig'
-lspconfig.util.default_config = vim.tbl_extend(
-  "force",
-  lspconfig.util.default_config,
-  {
-    autostart = false,
-    handlers = {
-      ["window/logMessage"] = function(err, method, params, client_id)
-          if params and params.type <= vim.lsp.protocol.MessageType.Log then
-            vim.lsp.handlers["window/logMessage"](err, method, params, client_id)
-          end
-        end;
-      ["window/showMessage"] = function(err, method, params, client_id)
-          if params and params.type <= vim.lsp.protocol.MessageType.Warning.Error then
-            vim.lsp.handlers["window/showMessage"](err, method, params, client_id)
-          end
-        end;
-    }
-  }
-)
+```vim
+lua << EOF
+require'lspconfig'.rust_analyzer.setup{}
+EOF
 ```
-
-## Individual server settings and initialization options
-
-See [CONFIG.md](CONFIG.md) for documentation and configuration of individual language servers.
-This document contains installation instructions for each language server, and is 
-auto-generated from the documentation in the lua source. Do not submit PRs 
-modifying CONFIG.md directly; CONFIG.md will be overwritten by docgen
-
-**You do not need to copy settings or init_options from this configuration into your config**
+For a full list of servers, see [CONFIG.md](CONFIG.md). This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup. Most language servers can be installed in less than a minute.
 
 ## Keybindings and completion
```

Comment:
for me this part is already advanced. I would lean to move it to another page and reference it here. The longer the root README.md, the less likely people will read I believe.

Reply:
by this you mean the following right? 

`For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup. Most language servers can be installed in less than a minute.`

Reply:
I meant moving the whole  `## Keybindings and completion` section to its own page or on the advanced page. 

As for the sentence you quote `Most language servers can be installed in less than a minute.` sounds unhelpful I would remove it. I would condense the next one with:
`For servers not on your system path (jdtls, elixirls) you must set `cmd` to the full path .` and maybe add a tag t `lspconfig-setup()-cmd-arg ` yo ucould reference here.

Reply:
Ooh maybe, but the issue is there are people who wanted this to be literally the first thing because if you copy/paste this you are basically done configuring 😆